### ISSUE TITLE
Verification gas limit and required prefund are too big in case of having paymaster

### ISSUE THREAD

**Issue by nhatcaoo** [eth-infinitism/account-abstraction#255]
The current implementation of the verification gas limit and required prefund when using a Paymaster can be problematic. The verificationGasLimit is used as a limit for the postOp call, and the calculation for requiredGas includes a multiplication factor of 3 if a Paymaster is being used. This can result in a significant amount of gas required for the transaction, which can be particularly burdensome if the verificationGasLimit is already high (e.g., for the first transaction to create a wallet) or if the wallet uses a complex scheme for verifying signatures.

`            uint256 mul = mUserOp.paymaster != address(0) ? 3 : 1;
            uint256 requiredGas = mUserOp.callGasLimit +
                mUserOp.verificationGasLimit *
                mul +
                mUserOp.preVerificationGas; `

One potential solution to this problem is to use a fixed value for the postOp gas limit instead of using the verificationGasLimit. This would help ensure that the gas required for transactions is more predictable and manageable for users, particularly when using a Paymaster.

In conclusion, the current implementation of verification gas limit and required prefund when using a Paymaster can be improved to make transactions more manageable for users, and using a fixed value for the postOp gas limit is one possible solution.

**Comment by drortirosh** [comment]
It is true that it can be high. We didn't want to put a constant, since there is no fixed value we know that will work always, on all chains.
Still, it is not charged, it is basically a limit. The paymaster pays eventually the exact amount required. 
It only means it has to have a deposit g enough.
